iambic/plugins/v0_1_0/aws/identity_center/permission_set/active_directory_utils.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+import json
+import socket
+
+import aiohttp
+from botocore.auth import SigV4Auth
+from botocore.awsrequest import AWSRequest
+
+
+async def alternate_list_users(session, directory_id, region):
+    domain = f"up.sso.{region}.amazonaws.com"
+    try:
+        ip_list = list({addr[-1][0] for addr in socket.getaddrinfo(domain, 0, 0, 0, 0)})
+    except socket.gaierror:
+        domain = f"up-sso.{region}.amazonaws.com"
+        try:
+            ip_list = list(
+                {addr[-1][0] for addr in socket.getaddrinfo(domain, 0, 0, 0, 0)}
+            )
+            assert ip_list
+        except socket.gaierror:
+            raise
```

Comment:
Worth pulling this out into a separate function since it's being used in multiple places?